### PR TITLE
Fixes bug with embedded images [SCI-1624]

### PR DESCRIPTION
**Pull request by ZmagoD**
https://biosistemika.atlassian.net/browse/SCI-1624

### CODE REVIEW

File: app/helpers/tiny_mce_helper.rb [hunk 1]
```diff
@@ -8,22 +8,24 @@ def parse_tiny_mce_asset_to_token(text, ref = nil)
       ids << img_id
       token = "[~tiny_mce_id:#{img_id}]"
       img.replace(token)
-      next unless ref
+      next unless obj
       tiny_img = TinyMceAsset.find_by_id(img_id)
-      tiny_img.reference = ref unless tiny_img.step || tiny_img.result_text
+      # next if check_parse_token_permissions(obj, tiny_img)
```
Comment (mlorb) [inline app/helpers/tiny_mce_helper.rb]: Why this is commented?

Comment (ZmagoD) [reply app/helpers/tiny_mce_helper.rb]: it's a banana

Comment (mlorb) [reply app/helpers/tiny_mce_helper.rb]: fix :banana:

File: app/helpers/tiny_mce_helper.rb [hunk 2]
```diff
@@ -56,4 +57,17 @@ def replace_tiny_mce_assets(text, img_ids)
...
+
+  def check_image_permissions(obj, img)
+    if obj.class == Step
+      img.step == obj
+    elsif obj.class == ResultText
+      img.result_text == obj
+    end
+  end
+
+  def check_parse_token_permissions(obj, img)
```
Comment (mlorb) [inline app/helpers/tiny_mce_helper.rb]: Where do you use this method?

**Pull request merged**